six/modules/c++/six.sicd/tests/test_read_sicd_from_vts.cpp

```diff
@@ -0,0 +1,321 @@
+{
+public:
+    TestScript(const std::string& pathname) :
+        mPathname(pathname)
+    {
+        std::ifstream file(mPathname.c_str());
+        std::string line;
+        while (std::getline(file, line))
+        {
+            lines.push_back(line);
+        }
+        originalSource = str::split(lines[1], " ")[1];
+    }
+
+    ~TestScript()
+    {
+        setSource(originalSource);
+    }
+
+    void setSource(const std::string& sourcePathname)
+    {
+        std::vector<std::string> sourceLine = str::split(lines[1], " ");
+        sourceLine[1] = sourcePathname;
+        lines[1] = str::join(sourceLine, " ");
+        write();
+    }
+
+private:
+    std::vector<std::string> lines;
+    std::string originalSource;
+    const std::string mPathname;
+
+    void write()
+    {
+        std::ofstream file(mPathname.c_str());
+        for (size_t ii = 0; ii < lines.size(); ++ii)
+        {
+            file << lines[ii] << "\n";
+        }
+    }
+};
+
+class ResultFile
+{
+public:
+    ResultFile();
+    ~ResultFile();
+    std::string pathname() const;
+    bool operator==(const ResultFile& rhs) const;
+private:
+    std::string mName;
+};
+
+ResultFile::ResultFile() :
+    mName(std::tmpnam(NULL))
+{
+}
+
+ResultFile::~ResultFile()
+{
+    std::remove(mName.c_str());
+}
+
+std::string ResultFile::pathname() const
+{
+    return mName;
+}
+
+bool ResultFile::operator==(const ResultFile& rhs) const
+{
+    std::string leftLine;
+    std::string rightLine;
+    std::ifstream leftFile(pathname().c_str());
+    std::ifstream rightFile(rhs.pathname().c_str());
+
+    while (std::getline(leftFile, leftLine) &&
+        std::getline(rightFile, rightLine))
+    {
+        str::trim(leftLine);
+        str::trim(rightLine);
+        if (str::startsWith(leftLine, "Program End Time"))
+        {
+            break;
+        }
+
+        if (str::startsWith(leftLine, "Program Start Time") ||
+            str::endsWith(leftLine, ".nitf"))
+        {
+            continue;
+        }
+
+        if (leftLine != rightLine)
+        {
+            return false;
+        }
+    }
+
+    return true;
+}
+}
+
+int main(int argc, char** argv)
+{
+    try
+    {
+        validateArguments(argc, argv);
+        const std::string sicdPathname(argv[1]);
+        const std::string outputPathname(argv[2]);
+
+        attachDESs(sicdPathname, outputPathname);
+        const std::string delimiter(sys::Path::delimiter());
+        const std::string testFile = sys::Path::joinPaths("six",
+            sys::Path::joinPaths("projects",
+                sys::Path::joinPaths("csm",
+                    sys::Path::joinPaths("tests", "test_script_sicd"))));
+        TestScript testScript(testFile);
+        testScript.setSource(sicdPathname);
+        std::string vts = sys::Path::joinPaths("install",
+                sys::Path::joinPaths("bin", "vts"));
+        sys::OS os;
+        if (!os.isFile(vts))
+        {
+            vts = vts + ".exe";
+        }
+
```

Comment:
Were you able to find a SICD 1.0?  If not, you can use one of the sample SICDs on the SIX wiki.